### DISCUSSION TITLE
Developer Meetings

### DISCUSSION THREAD

**Discussion by thabaum** [oqtane/oqtane.framework#3269] [Q&A]
I read in a release that there maybe a developer meeting for anyone who has ever contributed towards the project.  

1. Is there a date set like "third Saturday of the month"or some date set Quarterly?
2. Will this be done utilizing something like Discord or some other communication platform?

I would like to attend these once details become more available to at least listen in.

Thank you.

**Comment by sbwalker** [comment]
I am thinking of kicking this off next Tuesday, Oct 3. I will provide more details shortly.

**Reply by vnetonline** [reply]
Count me in please

**Comment by sbwalker** [comment]
I have created a list of ~20 of the most prolific contributors who submitted PRs to Oqtane in the past 2 years. This is a very diverse group in terms of time zones... so it is going to be very difficult to find a suitable time for everyone. 

I want to avoid weekends if possible, so I am proposing the first Tuesday each month (I have .NET Foundation meetings on the second and fourth week of each month already). So the first meeting would be Oct 3rd. 

I am also going to propose a fixed time of 12:00 ET (noon) which will be evening for the folks in Europe. I am also open to moving it earlier in the day if it works better for the majority of the group. 

As far as platform is concerned, I would prefer to use Microsoft Teams to host the meeting. At this point I am not interested in creating yet another channel for offline communication (ie. Discord) as it inefficient to have to monitor multiple channels and it will fragment the information even more. 

@hishamco, @leigh-pointer, @PavelVsl, @thabaum, @cnurse, @gjwalk, @vnetonline, @ADefWebserver, @iJungleboy, @alikoli, @tvatavuk, @W6HBR, @markdav-is, @HoenesDK, @Behnam-Emamian, @isaeed, @pepsinio, @dkoehnolms, @Rodien, @zzmzaizai, @nicpitsch, @rcpacheco

Please provide feedback on the proposal above. Note that I do not assume that everyone I have identified above is interested in participating in monthly meetings. If you want to opt out, that is completely fine with me as I realize that everyone has limited time.

**Reply by hishamco** [reply]
- Is it like ASP.NET / Orchard Core Standup Meeting?
- I'm not sure why everyone chose Tuesday specifically :)
- Is there a plan for the meetings or just status, please elaborate?

**Reply by iJungleboy** [reply]
This is great! and Tuesday should normally work for me as well - 12:00 ET is usually 18:00 in Switzerland, which fits really well. 

On Oct. 3rd I won't be able to participate as I'm on holidays in Korea, but count me in in November.

**Reply by ADefWebserver** [reply]
Works for me

**Reply by sbwalker** [reply]
@hishamco it will be a meeting to share current status, future plans, blockers, etc... The meetings will have an agenda and will be 1 hour in duration.

**Reply by tvatavuk** [reply]
- Thank you for organizing this and considering the diverse time zones of the contributors. The proposed time on the first Tuesday of each month at 12:00 ET sounds great to me. I believe it's a good compromise given the circumstances.
- Using Microsoft Teams as the platform for the meeting is also a good choice, as it's widely used and reliable.
- Thank you for taking the initiative, and I look forward to our first meeting on Oct 3rd.

**Reply by sbwalker** [reply]
@hishamco good point about the .NET Community Standup... perhaps it would be better to schedule this meeting for Wednesdays?

**Reply by leigh-pointer** [reply]
@sbwalker 
Tuesday or Wednesday is good for me.

**Reply by hishamco** [reply]
The means 4 OCT is the meeting day, could you please create a Teams Link that what we did in Orchard Core, so it makes easy for everyone to join

**Reply by sbwalker** [reply]
@hishamco I will share a Team link once the date/time is finalized and others have had a chance to speak up.

**Reply by hishamco** [reply]
Please consider there are some outside America & Europe (Asia), like my situation I'm in GTM +3

**Reply by sbwalker** [reply]
@hishamco yes, I am aware. @vnetonline is in Australia.

**Reply by W6HBR** [reply]
Date/time works for me (Tuesday or Wednesday).

**Reply by vnetonline** [reply]
Tuesdays also works for me looking very much forward to the monthly meetings

**Reply by alikoli** [reply]
I won't be able to attend this meeting, I hope to be invited for the next one.

**Reply by pepsinio** [reply]
Proposed Date & Time are suitable for me as well.

**Reply by markdav-is** [reply]
Sorry I missed the first one.  I'll make it next time.  9AM Tuesdays PT works good for me.  I'll put it on my cal.

**Comment by nicpitsch** [comment]
Thanks for the invitation

**Comment by sbwalker** [comment]
@sbwalker, @hishamco, @leigh-pointer, @PavelVsl, @thabaum, @cnurse, @gjwalk, @vnetonline, @ADefWebserver, @iJungleboy, @alikoli, @tvatavuk, @W6HBR, @markdav-is, @HoenesDK, @Behnam-Emamian, @isaeed, @pepsinio, @dkoehnolms, @Rodien, @zzmzaizai, @nicpitsch, @rcpacheco

The kickoff meeting will be on Wed, Oct 4 at 12:00-1:00 PM ET. You can connect to the meeting using Microsoft Teams.

The agenda will be:

- Intros
- .NET 8 plans
- Roadmap priorities
- Marketplace status
- Q&A
- Confirm next meeting date/time

I look forward to a productive session.

**Comment by sbwalker** [comment]
@sbwalker, @hishamco, @leigh-pointer, @PavelVsl, @thabaum, @cnurse, @gjwalk, @vnetonline, @ADefWebserver, @iJungleboy, @alikoli, @tvatavuk, @W6HBR, @markdav-is, @HoenesDK, @Behnam-Emamian, @isaeed, @pepsinio, @dkoehnolms, @Rodien, @zzmzaizai, @nicpitsch, @rcpacheco

Reminder of Oqtane Developer Monthly Meeting today at 12:00-1:00 PM ET (4:00-5:00 UTC)

**Reply by ADefWebserver** [reply]
I'll be there for the .NET 8 plans...

**Reply by hishamco** [reply]
@sbwalker where's the Teams link?

**Reply by sbwalker** [reply]
@hishamco I posted the Teams link last week and then removed the post because it is a private. Each person who is invited to the meeting would have received an email notification from GitHub. I will post it again now (and then remove the post - so please check your email)

**Reply by sbwalker** [reply]
@hishamco please let me know if you received the email from GitHub

**Reply by hishamco** [reply]
I will check my Yahoo coz I didn't sign in frequently

**Reply by markdav-is** [reply]
Sorry missed the first meeting 9AM PT Wednesdays will be hard, but I'll make it happen next week.

**Reply by sbwalker** [reply]
@markdav-is these will be monthly meetings - the next meeting will be Nov 1

**Comment by nicpitsch** [comment]
I'm absent, outside. Sorry
Best regards Nicolo

Shaun Walker ***@***.***> schrieb am Mi., 4. Okt. 2023, 17:52:

**Comment by hishamco** [comment]
We're LIVE now!!

**Comment by thabaum** [comment]
Thank you @sbwalker for putting time and effort into this meeting.  It was awesome experience getting to meet you and  everyone else who attended.  I look forward to helping in any form address issues as I can.  We touched on some big issues going forward and I believe everyone was as thrilled as I was.

On some of these topics maybe we can form a discussion here on things like .NET 8 enhancements, documentation and scalability that can be ongoing threads to be further addressed.  

My apologies to everyone if my etiquette was not inline as this was my first "developer" and "team" meeting.   I didn't have time to notice all the raise hand buttons and so forth until I seen one used. I have set these up for clients in the past many times, but never attended one however believe it or not.  I was able to learned from it and look forward to being more prepared next meeting.  It will be a memorable ice breaker to be a part of this meeting and hear how everyone was utilizing Oqtane in production.  

Few things I wanted to give as feedback but didn't want to take too much valuable time discussing towards the end as I felt I could share this more here relates to:

Documentation:

https://github.com/oqtane/oqtane.docs

This is where the docs should be correct?  And from here the website can build out the pages.  You mention all the videos and blogs we have available, but like I pointed out they are a bit scattered and time chasing is valuable time to anyone trying to get a project started.

What I feel as a community we can all chip in on is like submitting PR's here is getting more pushed into docs, even if it is a link to the blog post(s) relating, videos produced and other valuable information. It doesn't have to be perfect to start as we gather all the valuable resources online into one place categorized.  

My idea was to chip in on this when I have time time as I consume them and try to create PR's that get these resources organized at least to start.  Maybe later down the road when financial resources become available it can be laid out in a more professional way.  But for now just getting things grouped together in one place by topic would be huge start to me.

Marketplace
Module Development
Theme Development
Installations
Submitting Feedback
Contributing Source Code

When it comes time I want to go right to a source that explains the process for when this product to enter the market place and fully understand basic licensing when it is ready for production use. So I believe a focus on getting these resources organized on the website is crucial to our ability to make a choice to work that direction.  Limiting the chasing of information is the bottom line.   And as you said many times the information is there, we just need one place to pull it all up.  And not just "links" but video player embedded you can popup if needed for videos relating along with the blogs as a popup that keeps us on the same page. 

You can write the documentation in blogs, but the blogs need categories in your blog module which would help as well.  I was going to produce a PR for that at some point, but you might want to do it sooner now that you have more blogs to sort through along with a search feature.

Also maybe put a video slider of all your YouTube videos on the homepage to help new people get to information they may be looking for to determine if Oqtane is the right solution for their project.

.NET 8:

For .NET 8 I do plan to explore all the features this month now that they finally have them all packed into the latest preview.  I want to review their router and Oqtanes router to see what might need merged to produce results to help leverage the new features around it, but I would not say in any way that is a deal breaker here for Oqtane.  Oqtane is very unique.  It is a nice feature but maybe it is fine tuned in .NET 9/10 (Microsoft skips version 9 if I recall?)

The main thing that separates this framework from all the others like Wordpress or DNN as far as a CMS goes, is this will create a mobile app as well.  I think the .NET Maui part of this framework needs to be more highlighted as I think it is why I am here!  I would of jump ship entirely to javascript if this wasn't a part of .NET Core and what Oqtane can accomplish.

When I tell potential clients I can create a mobile app using a single code base along with custom modules rapid development their eyes light up.  The thing is, I am my biggest client.  

Scalability:

I have a project that will need to scale.  My old school terminology dates me but firing up a few VM's on different servers in different locations is important to my company project.  It was great to get an insight on where Oqtane is heading next in respect to this and the community support from other developers in this area.  I look forward to help with testing this when these features come to light.

I did have a number of other things I wish I could of discussed but only so much time in an hour.  It was nice to more clearly speak to you on topics like documentation and migration support.  

I have a module I can put out now into the Marketplace that I am working on.  Pretty much all the modules I am creating for this larger project will be able to become commercial if I feel it would be worth supporting for the Oqtane community.  I have one I know I always looked to have on my sites I can potentially put out through my company.  

I am currently a one person show... so I am shocked I spend the time I do to type all this feedback out.  As you should tell I am very passionate about a project I am cutting the line from DNN even after buying a huge selection of modules and spending countless hours developing to migrate to something like Oqtane.  I have not done it yet, but things like easier migration of users/pages/sites is a big effort for a small company with a single developer trying to tackle so many things by New Year which is my goal.  My company will probably be one of the "Professional" version subscribers for the added value support at some point in time when revenues meet supporting it, although we maybe sending Oqtane more support than we are asking for.

I am pleased to hear Oqtane will NOT follow DNN's path as an added feature business and more a professional support service subscription.  At the same time all the features DNN had in the professional version where the ones I was after.  Purchase of active Social project excited me at first but then bummed me out to see it bought out and then go for huge amounts of money to utilize leaving the community with a broken down version that never got enhanced.  The provider for Facebook got killed and never supported destroying my business I created that used it.  At that point I was a DNN user more than a contributor and I don't think anyone realized I was a programmer in the community.  Then I learned how to put PR's in Github and was offered a spot on one of the community teams there after a number of contributions came out of the woods by me.  Then .NET shifted to .NET Core... what a chain of events.  A nightmare I don't wish to repeat. 

For a startup with shoe string budget the cost and risk for what it wanted yearly to keep using it was impossible. I felt a lot of ideas I had made it to the pro version I shared and not in the community edition as well.  So I backed off those contributions.  A personal goal of mine is to instead of building out all the modules in DNN that needed updated for the community, build them here in Oqtane once it matured enough to have a clear road ahead of me.  If I had one other person that could contribute and review things with me they would be available now open source to the Oqtane community.  

One thing that disappoints me about Oqtane and DotNetNuke is how hard it is to connect with other developers who may be interested in taking on a collaborative effort in this arena.  I have tried to create a Discord for some sort of place developers that are more freelancers can hang out and make connections like this possible but it was dead and never taken serious.  So from that perspective I came to the conclusion everyone is out for themselves and it is dog eat dog as everyone competes for supporting clients that use the frameworks.  This is most likely true for any open sourced project, not just DNN or Oqtane.

When it comes time I want to go right to a source that explains the process for when this product to enter the market place and fully understand basic licensing when it is ready for production use. So I believe a focus on getting these resources organized on the website is crucial to our ability to make a choice to work that direction.  Limiting the chasing of information is the bottom line.   And as you said many times the information is there, we just need one place to pull it all up.  And not just "links" but video player embedded you can popup if needed for videos relating along with the blogs as a popup that keeps us on the same page. 

You can write the documentation in blogs, but the blogs need categories in your blog module which would help as well.  I was going to produce a PR for that at some point, but you might want to do it sooner now that you have more blogs to sort through along with a search feature.

This was a recap of some thoughts I had during the meeting for some additional feedback while it is fresh.  My apologies again for this long message, I will work on keeping them shorter.

Sounds like first Wednesday of each month 12-1 EST is set for future meetings which is awesome!

Thank you everyone for letting me speak when I did and contributing to this project, I  have always considered everyone here as colleagues like fellow employees.

**Reply by ADefWebserver** [reply]
Regarding Documentation

1. I have tried to contribute to the [Github repo](https://github.com/oqtane/oqtane.docs) in the past and it is way too much trouble to contribute. That is why I stopped contributing to it.
2. We are a CMS (I know 'don't call it a CMS' but still...) and we should be 'dogfooding' this task in Oqtane. If a new Module needs to be created, this is what we should do IMHO

**Reply by thabaum** [reply]
I agree @ADefWebserver and to make something like documenting less cumbersome would be great way to showcase Oqtane.  Reading up more on "dogfooding" and documentation I see [https://developers.cloudflare.com/](https://developers.cloudflare.com/ )uses this approach for their documentation.

[https://blog.cloudflare.com/identifying-content-gaps/](https://blog.cloudflare.com/identifying-content-gaps/) More information on why and how they use the same approach.

It would be nice to have an Oqtane.Documentation module able to brand for assisting with the application end use for administrators and end users, that can be improved upon easily with more documentation relating to the custom features that application has.

I think we are over the fear of CMS name calling as Oqtane is officially now considered a CMS and Application Framework built specifically for developers and not the average user.  It is open to be extended to be a CMS for non-technical users but can be stripped down to just a modular app.

I would like to see an attempt making a prototype module we can all review and more easily contribute to for the "Oqtane CMS and Modular App Framework" developers.  This module if created since not a part of the actual framework should be able to use more third party libraries I am guessing or would it hold the same restrictions?

**Comment by vnetonline** [comment]
The meeting was at 3am for me .... tried to stay awake but didn't make it was the meeting recorded???

**Reply by thabaum** [reply]
@vnetonline Sounds like you need to go to bed early like 7pm and wake up at 3am, then take a quick nap before work?  Maybe next month as I am happy to hear these are monthly and set to a day I can easily schedule.  Given I don't have to put out any fires so to speak I will attend all I can.  Look forward to meeting you at one!

**Reply by sbwalker** [reply]
@vnetonline my apologies... I neglected to record the meeting :( I can send you the slides by email.

**Reply by iJungleboy** [reply]
@sbwalker I'm just back from my 3 week vacation, so I missed it too. Could you send me the slides as well, thx

**Comment by markdav-is** [comment]
@sbwalker is there a meeting this week?  If so, I'd like to get it on my cal.

**Reply by sbwalker** [reply]
@markdav-is the meetings are monthly - the next meeting will be  Nov 1

**Reply by ADefWebserver** [reply]
@sbwalker Will you send a Teams invite?

**Reply by markdav-is** [reply]
thanks.  do you know what time PST?

**Reply by sbwalker** [reply]
@markdav-is the meetings were established as the first Wednesday of every month at 12:00-1:00 PM ET
@ADefWebserver I do not have email addresses for the majority of the attendees - perhaps people could send me their email using https://www.oqtane.org/#contact

**Comment by sbwalker** [comment]
Reminder of the Oqtane Developer Monthly Meeting on Nov 1 (tomorrow) at 12:00-1:00 PM ET (4:00-5:00 UTC)

@hishamco, @leigh-pointer, @PavelVsl, @thabaum, @cnurse, @gjwalk, @vnetonline, @ADefWebserver, @iJungleboy, @alikoli, @tvatavuk, @W6HBR, @markdav-is, @HoenesDK, @Behnam-Emamian, @isaeed, @pepsinio, @dkoehnolms, @Rodien, @zzmzaizai, @nicpitsch, @rcpacheco

I requested that the folks above send me their email addresses using this form: https://www.oqtane.org/#contact (bottom of page) - however at this point I only have 5 email addresses. Please use the form so that I can add you to the invite.

Tomorrow's meeting will be primarily focused on .NET 8, as there is a lot to explain about this topic. I will remember to record the meeting this time.

**Reply by vnetonline** [reply]
@sbwalker just sent through my contact details

**Reply by hishamco** [reply]
The same teams link or you will send it through the contact form

**Reply by sbwalker** [reply]
@hishamco it will be the same Teams link - I also added your email from your GitHub profile to the meeting invite

**Reply by hishamco** [reply]
Thanks, see you then

**Reply by Rodien** [reply]
@sbwalker I won't get home in time for the meeting this time. I'll make sure to check the recording.

**Reply by thabaum** [reply]
Great meeting, sounds like consensus on .NET 8 is sit back relax, get fat and be happy with Oqtane 5.0 with nothing to change here!

@sbwalker To be more clear in my final message what I was trying to say with versioning was to set 5.0 to .NET 8 to follow the roadmap as explained but typed it out wrong hit enter trying to correct my message and it went out early with 4.1 when I meant to say 5.1...  Sounding dumb and incompetent again I was not trying to make the versions not match the releases of .NET Blazor.  Bottom line what I was trying to say as I typed was 5.x could be the newer version if no solution is created for Oqtane prior to 5.0 release relating to refactoring logic to the new render modes due to complexities.  Or it could even be later in version like 6 for the SSR + CSR support.

The consensus was "give up on .NET 8 SSR + CSR an leave Oqtane as is"  which to me sounds like which is the easy way out, maybe not the solution however.  The problem i see is down the road with third party support and supporting features in .net 9 or 10 when things are actually closer to being "Blazor United" since this marketing term was said to be premature and is not intended to be used to describe .NET 8 after watching a community standup back in May 2023 since it makes people think there is a huge change to the core framework when there is not.  

"Blazor Full Stack" or just "Blazor" was the new marketing behind the next version .NET 8 as you will not see Microsoft call it "Blazor United" since it does not unite the mobile app side of things as of yet into one project which is the ultimate goal long term.

Correct me if I am wrong, please! [https://www.youtube.com/watch?v=MaLjiR9YSbs&list=PLdo4fOcmZ0oX-DBuRG4u58ZTAJgBAeQ-t&index=31](https://www.youtube.com/watch?v=MaLjiR9YSbs&list=PLdo4fOcmZ0oX-DBuRG4u58ZTAJgBAeQ-t&index=31
) 12 minutes into video and 1:13 into it again talks about this again.  I am pretty sure it was  renamed "Blazor Full Stack" but I would have to dig that up possibly in the next video or two.  .NET 8 is not "Blazor United" to be clear here which is confusing if used and was dropped to avoid this confusion.  .NET 8 is more or less just SSR + CSR.

Oqtane will work fine as it is and one goal for the roadmap of Oqtane 5 is support for server farm or web farm or whatever term is used today for multiple servers (instances of Oqtane) running one site.  Hopefully all of that work does not need to refactored once things are caught up to .NET 8 SSR + CSR rending in Oqtane.  This along with issues in third party support for the UI components as discussed.

The problem is focus on my missed typed information with following the .NET 8 version with Oqtane version and less focus on what was trying to be said.  I had my mic to talk but there where only seconds left on the stage... just tossing out some feedback that wasnt 100% taken into context during the end of the meeting.

I provided testing and info along with a couple PR's to help with .NET 8 and Oqtane 5 best I could.  I have no plans on diving any further into this now that I have heard what I have.  It has been one heck of a ride from DNN 4/5 to Oqtane 4/5, thanks again to all contributing in both worlds from everyone!  It has been a real pleasure getting to know a few of you along the way!  Best wishes and luck... cheers!

**Reply by sbwalker** [reply]
I believe the consensus was "Release Oqtane 5.0 for .NET 8 but focus on the existing Blazor Server, Blazor WebAssembly, and Blazor Hybrid hosting models". This provides a very solid foundation for Oqtane developers who want to use the .NET 8 LTS release. Continue working on the new Blazor SSR approach (probably as a separate branch) to create a POC and do an impact/feasibility assessment (as there are still many issues to iron out). These efforts may or may not materialize in a 5.1 release - it depends on the outcome of the POC.

**Comment by leigh-pointer** [comment]
Anyone got the TEam meeting ID

**Reply by hishamco** [reply]
Your email?

**Reply by leigh-pointer** [reply]
@sbwalker  Received no invite

**Reply by hishamco** [reply]
The meeting is almost finish, you can contact @sbwalker later to join us on the next meeting

**Reply by sbwalker** [reply]
@leigh-pointer verifying if you have the invitation for the meeting today at 12:00 PM ET  (I know you missed the meeting last month). Your email address is included on the invitation.

**Reply by leigh-pointer** [reply]
@sbwalker received no email today?

**Reply by sbwalker** [reply]
the recurring meeting invite was sent out last month

**Reply by sbwalker** [reply]
@leigh-pointer it was sent last month to your @outlook.com email address

**Reply by leigh-pointer** [reply]
@sbwalker I have that email will join now and wait for it to start

**Reply by sbwalker** [reply]
it starts in 1 hour

**Reply by thabaum** [reply]
Great meeting! Thanks for all the insights, efforts and feedback towards everything from everyone.  Oqtane 5.1 sounds like is in the works and if the rubber can meet the road unlocking more features for Oqtane.  As pointed out there are some complications yet to be ironed out in .NET 8 as I have found out myself from trying to incorporate using the new features on components inside new applications with different project setups in Visual Studio.

Oqtane running with global interactivity still got faster and even if it stayed as is the application framework itself is still a solid foundation for any application.  

It was interesting to hear about the changes with the Blazor Hybrid hosting model settings within Oqtane.  It has me a bit concerned as some of the features Oqtane used from .NET are now more granular so to speak with the hybrid option to set.  I look forward to hearing more about this as it sounded like this was on by default on all Oqtane hosting model settings?  I am interested in knowing if this relates to allowing the Maui application to connect for a "Hybrid" Mobile/Web application.

This Microsoft doc has been updated recently to help explain the different render modes:

[https://learn.microsoft.com/en-us/aspnet/core/blazor/components/render-modes?view=aspnetcore-8.0](https://learn.microsoft.com/en-us/aspnet/core/blazor/components/render-modes?view=aspnetcore-8.0)

But there are still some pretty big gaps in my complete understanding of how and why somethings work and others do not.  It almost adds an entire new level of self doubt while trying to be concrete and sound in a decision of which to use.  Seems like enhanced forms is also a big focus for .NET 8 features as it changes things a little bit all over the place if that makes sense when it comes to rendering.

It's like Microsoft added a few new pieces to the chess board.  Maybe faster and more powerful, but definitely more to think about and be aware of.

Sounds like we may get one or two good [**Oqtane Blog**](https://www.oqtane.org/Blog) topics out of that meeting with the Marketplace and other talking points potentially!?  I look forward to reading them when they become available.

Some feedback on the Oqtane Blog since it is our source for documentation should be included in the [**Oqtane Docs - Articles**](https://docs.oqtane.org/articles/index.html) pages.  It would also be helpful to categorize the blog posts now there are many into sections that represent areas of the documentation so you can locate the relating blogs of interest easily if available.  This would take an investment of a PR to add category/subcategory feature to the blog module and some time adding the links to this articles section of the docs by category of blog posts relating to framework help and support.

It would be a bit of a task to start, but once caught up it would be a wealth of links to information that might help someone get unstuck and not think of searching that area and instead finds the help by browsing the links to the organized blog posts in this articles section.

Just an idea if the two can be connected to help enhance guiding users and developers of Oqtane without adding too much overhead.  Right now this docs page has some [BlazorHelpWebsite.com](https://blazorhelpwebsite.com/ViewBlogPost/15) articles which is great.  It would be helpful to have the blog articles get categorized with categories used both on the site and the docs articles section to locate the administration, release information and developer relating blog/site articles.

This would give a valuable back link to a source of information already provided that could help figure something out that has been discussed or find new ways to enhance their Oqtane Framework development or administration experience.  8 pages and counting of blog articles, the search feature is nice, however some general categories would be an enhancement.  

Categories such as "Development", "Administration", and "Release Information" along with "General" would help find information on topics I know get asked about more than once.  Probably not all blog articles need to be listed here.  Just the ones that direct developers or administrators to helpful blog articles.

A link to the blogs page from the Oqtane docs articles page would at the least be helpful and then add the "category" option for filters and/or "tags".  Since the Oqtane Blogs seems to be one of the best sources for information it should be included in the Articles Section Homepage in some form is a minimum suggestion and on the homepage or index "Welcome to Oqtane Articles".

I like the blog articles, but wish they were more accessible from docs.  I can see the overhead between the two so the end call is not up to me, but I believe at least a link going from Oqtane Docs Articles section to the Oqtane Blogs home page would be a huge enhancement for the docs alone.  It would help someone understand they should review or search the Oqtane site blogs for answers to issues as well as review the documentation pages.

The Oqtane blogs will be thought of as more the additional "docs" it is currently being utilized as if included somehow into the articles section.  Keep it simple so it functions with ease.

Love the progress and community support along with the overall direction of things staying on track and in focus.

Cheers!

**Reply by sbwalker** [reply]
@thabaum the https://docs.oqtane.org/ site is generated from the Github repo. Anyone can add content (ie. add sections to organized, add new links, etc...) by submitting PRs to that repo. At one of the previous Developer meetings we discussed why nobody has ever submitted a PR to this repo since it was created by @iJungleboy ~2 years ago - and the consensus seemed to be that the tooling was too difficult to understand. Currently it works the same as any static site generator (ie. HTML, CSS, Git) but I guess people are hoping for a more streamlined experience. Creating a docs app would require some effort - so in the short term I would think that submitting PRs to this repo is the quicker approach.

**Reply by iJungleboy** [reply]
@thabaum The docs site uses DocFX which is the de-facto standard in the .net ecosystem. 

It uses Markdown + auto-generated code documentation based on the XML-docs in the code itself. 

I think changing the approach would not make sense, and would not really change the load of contributions either ;).

**Reply by thabaum** [reply]
@iJungleboy I believe we should investigate how we can increase contributions further even if just simple ones at this point.  Oqtane has matured to a level things need to be updated to reflect the latest changes in the framework itself.  I get it that its a starting point for anyone creating documentation.  I almost feel "dogfooding" would be like as an idea turning the Oqtane Blog module into an Oqtane Knowledge-base type of module which users can contribute to, or we use Oqtane Docs to do it but we need a place that allows this in some form or degree if you know what I mean.  I have put ideas up as issues but not getting any feedback from anyone on them to understand what can or can't be done to get a feel for right/wrong approaches to properly documenting the Oqtane Framework.  I can maybe dedicate some time towards this if I had a bit more direction and response in this area.  Just some feedback as I try to understand how I may be able to help myself and others in this area of the product.

**Reply by ADefWebserver** [reply]
I like the idea of a **Oqtane Knowledge-base type of module**. I think I may be the person who has written more on Oqtane than anyone else at this point and I can assure you that using markdown, github, or any sort of 'build process' is a non-starter for me. 

If there were a **Oqtane Knowledge-base type of module** I would immdeiatly do a PR and add [WilderMinds.MetaWeblog](https://github.com/shawnwildermuth/MetaWeblog) to it so I can continue to use [Windows Open Live Writer](https://apps.microsoft.com/detail/9NBLGGH5279M?launch=true&mode=full&referrer=bingwebsearch&ocid=bingwebsearch&hl=en-us&gl=US) to post my content.

**Reply by ADefWebserver** [reply]
Here is the fastest way:
1) Alter the [Blogs module](https://github.com/oqtane/oqtane.blogs) to support multiple bloggers in a single instance on a single page
2) Add [WilderMinds.MetaWeblog](https://github.com/shawnwildermuth/MetaWeblog) to it so people can use [Windows Open Live Writer](https://apps.microsoft.com/detail/9NBLGGH5279M?launch=true&mode=full&referrer=bingwebsearch&ocid=bingwebsearch&hl=en-us&gl=US) to post content

This will allow the team to easily post helpful content. This is how DotNetNuke handled 'documentation' for years. To this day there is no online up to date formal documentation. Basically Blogs and Forums is how most projects get by.

Oh did I mention that *we need a forum module?* :)

**Comment by sbwalker** [comment]
A summary of the recent Monthly Developers Meeting was posted here: 

https://www.oqtane.org/blog/!/76/oqtane-developers-meeting-dec-2023

**Comment by thabaum** [comment]
Great meeting thanks to everyone who attended and shared what they could.

Community Side Project:
Sounds like we may be heading towards creating a wiki/blog with live writer as an additional way to find help with rich content support documents.  This can also serve as a way for other contributors to find what should be included in the DocFX for those that wish to continue with that project.  I think we can have both worlds and they can complement each other.

Roadmap:
5.1 Enhances Oqtane Framework to include Global Interactivity (current version supports this now but enhanced for the .NET 8 architecture) 
5.2 Introduces "static" rendering if results prove worth driving down this path.

Very cool and exciting to hear about all the enhancements included in 5.0.1 as well as it resolved an issue I wanted to bring up... Big thanks for all the additional love and support!

Happy New Year Oqtane!

**Comment by sbwalker** [comment]
Summary of January Developer Meeting posted here: https://www.oqtane.org/blog/!/79/oqtane-developers-meeting-jan-2024

**Reply by sbwalker** [reply]
@ADefWebserver in regards to your question of why we are prioritizing the integration of the new Blazor approach in .NET 8, I think this item in the Oqtane Philosophy (https://www.oqtane.org/blog/!/20/oqtane-philosophy) explains it fairy well:

"Microsoft Alignment

Although there are cases where it makes sense to come up with original solutions to development problems, it is generally an industry best practice to take advantage of the native solutions which are offered by the providers of the platform. This alignment helps to ensure that the approach is well validated and will be supported in the future. This focus on consistency also minimizes the time it takes for developers to come up to speed on your framework because they are already familiar with the approach based on their previous development efforts. Constantly focusing on creative new solutions, some of which may be slightly better than what Microsoft provides, can become a barrier to entry for new developers and a maintenance headache for the platform in the long term. Oqtane will utilize the Microsoft recommended approach as much as possible and only be creative in the areas where Microsoft does not offer a solution. A good example of this is that Oqtane uses Nuget for its packaging and deployment strategy as opposed to creating its own proprietary approach."

I will be sure to reference this in the blog post I intend to write to elaborate on this topic.

**Comment by sbwalker** [comment]
@pyramidsbuilder based on your PRs over the past month, can you please send an email to support@oqtane.org so that I can add you to the monthly Oqtane developer meeting invitation (which is happening next week)

**Reply by pyramidsbuilder** [reply]
Email sent thank you very much @sbwalker

**Comment by sbwalker** [comment]
Good meeting today... I will be posting a summary of the discussion on the Oqtane blog.

**Reply by sbwalker** [reply]
Meeting notes here: https://www.oqtane.org/blog/!/83/oqtane-developers-meeting-mar-2024

**Comment by sbwalker** [comment]
@mdmontesinos  based on your PRs over the past month, can you please send an email to [support@oqtane.org](mailto:support@oqtane.org) so that I can add you to the monthly Oqtane developer meeting invitation (which is happening this Wednesday, April 3 at 12:00 PM ET)

**Reply by mdmontesinos** [reply]
Will do, thanks for the invite! Although I'm not sure if I'll be able to attend.

**Comment by thabaum** [comment]
Great meeting Oqtane developers!  Also my hat is off to everyone for all the hard work to generate this latest release.  Hand full of stabilization issues to follow.  Sounds like we will be heading in the direction of some future roadmap enhancements while continuing to improve the core of Oqtane.

A Discord channel has been proposed again.  I believe anyone that wants to manage this for the Oqtane community can take this on and maybe allow a few others interested possibly to help maintain it.

It was nice to see the maintainer with less on his plate (@sbwalker ) allowing more community discussions to exist.

Some interesting topics discussed included use of AI in Oqtane and iOS deployment issues relating to the tools used to accomplish this more than Oqtane itself being an issue.  It sounded to me like a discussion could be generated here to discuss these topics further and pull from it what is found worthy to note into documentation.

Look forward to the 5.1.1 release.  I did upgrade my production test site successfully, however i did notice 3 errors in the System Info logs I will be reviewing, one having to do with Anti-Forgery Token along with a couple others I will investigate when I have time.  I believe one also had to do with serialization.  I will report these once I know what to report... these may have been dealt with already in recent fixes merged.

I also had one other issue that made the navigation UI controls not actually get set to inline in the theme creator theme I was using on the test production site.

We also discussed some known issues in Oqtane 5.1 so be sure to review the minutes as they will provide some valuable info.

Other than that it looks like things are looking great going forward into the future after two months of community beast mode by our developing contributors to get to this point.   My hat is off to you all!

Cheers!

**Reply by mdmontesinos** [reply]
Was the meeting recorded and uploaded somewhere??

**Comment by sbwalker** [comment]
@ohba-ikuo, @ijaz-saeed, @marceloatoledo  based on your recent contributions, can you please send an email to [support@oqtane.org](mailto:support@oqtane.org) so that you can be added to the monthly Oqtane Developer Meeting invitation (which occurs on the first Wednesday of each month at 12:00 PM ET - next meeting is tomorrow Jul 3)

**Reply by hishamco** [reply]
16:00 GTM right?

**Reply by sbwalker** [reply]
correct
